packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -202,8 +221,9 @@ export enum SubnetType {
    * Normally a Private subnet will use a NAT gateway in the same AZ, but
    * if `natGateways` is used to reduce the number of NAT gateways, a NAT
    * gateway from another AZ will be used instead.
+   * @deprecated use `PRIVATE_WITH_EGRESS`
    */
-  PRIVATE_WITH_NAT = 'Private',
+  PRIVATE_WITH_NAT = 'Deprecated_Private_NAT',
```

Comment:
Does this have the potential to cause breaking changes for users?

Reply:
I do not believe it will and these constants have been through a rename cycle before in v1.
You can see an earlier PR which renamed it originally
https://github.com/aws/aws-cdk/pull/19320

This actually caused an issue with VPC resources being recreated which you can see was ultimately fixed in
https://github.com/aws/aws-cdk/pull/21140

I have added similar tests to what was done in that last pull requests to get the same behavior.